### ISSUE TITLE
Misleading functions in `clash-prelude-hedgehog`

### ISSUE THREAD

**Issue by martijnbastiaan** [clash-lang/clash-compiler#2138]
E.g., `genSigned` can generate results outside of its given range:

https://github.com/clash-lang/clash-compiler/blob/690fbea81758444d752e750964141d3f790677c8/clash-prelude-hedgehog/src/Clash/Hedgehog/Sized/Signed.hs#L28-L34

I think this is misleading at best. We probably want to inspect the given range and generate _its_ extreme values.

**Comment by alex-mckenna** [comment]
Yes, that seems reasonable. Do you want to make a PR or should I, it's just a case of changing

```haskell
minBound ~~> Range.lowerBound range
maxBound ~~> Range.upperBound range
```

**Comment by martijnbastiaan** [comment]
Go for it

**Comment by martijnbastiaan** [comment]
We've released [v1.6.3](https://github.com/clash-lang/clash-compiler/releases/tag/v1.6.3), which includes a fix for this issue.
